### PR TITLE
Create testinfra tests

### PR DESCRIPTION
**Pull request by gndrmnn**
| Role | Merged into main #1238 | Note |
| --- | :---: | --- |
|  ``adminer`` | × |  :heavy_check_mark:  |
| ``auditd`` | × |  :heavy_check_mark: <br> Actual adjustment content is not checked. |
| ``bird`` | × |  :heavy_check_mark:  |
|  ``cephclient``   | × |    :heavy_check_mark:  |
| ``cgit`` | × |   :heavy_check_mark: |
| ``chrony`` | × |  :heavy_check_mark: <br> Actual config content is not checked. |
| ``clamav`` | × |  :heavy_check_mark: <br> Actual config content is not checked. |
| ``containerd`` | × |  :heavy_check_mark: |
| ``dnsdist`` | × |  :heavy_check_mark: |
| ``docker`` | × |  :heavy_check_mark: <br> Docker registry login is not checked. |
| ``fail2ban`` | × |  :heavy_check_mark: |
| ``falco`` | × |  :heavy_check_mark: |
| ``frr`` | × |  :heavy_check_mark: |
| ``hddtemp`` | × |  :heavy_check_mark: |
| ``homer`` | × |  :heavy_check_mark: |
| ``journald`` | × |  :heavy_check_mark: |
| ``keycloak`` || :heavy_check_mark: :question: <br> (1) Inexplicable: FAILED tests/keycloak.py::test_keycloak_service[ansible:/molecule-keycloak] <br> (2) Use of mariadb within default variable "keycloak_galera_backend_enable: true" seems to be obsolete. Cf. ``keycloak.py`` 63-80. |
| ``lldpd`` | × |  :heavy_check_mark: |
| ``manager``||  :heavy_check_mark: :exclamation: <br> Testing ara_server_mariadb_tag < "11.0.0" not working. Cf. ``manager.py`` 184-188. |
| ``metering`` || :heavy_check_mark: <br> Added missing Ansible handler in role.  |
| ``netbox`` | × |  :heavy_check_mark: |
| ``netdata`` || :heavy_check_mark: <br>Test refactored |
| ``nexus`` | × | :heavy_check_mark: <br> Role debugged: cf. ``config.yml``, ``initialize.yml`` and ``docker-compose.yml.j2``  |
| ``openldap`` || :heavy_check_mark: <br> Error ``failed to read /opt/openldap/configuration/udm-rest.env: line 7: unexpected character \"/\" in variable name \"ldap/base=dc=osism,dc=local\"`` fixed by modifying ```udm-rest.env.j2```. |
| ``openstack_health_monitor`` | × |  :heavy_check_mark: |
| ``openstackclient`` | × |  :heavy_check_mark: :question: <br> This is running against main branch instead of our branch?! |
| ``osquery`` || :heavy_check_mark: <br> ~~Todo: Race Condition~~ |
| ``phpmyadmin`` | × |  :heavy_check_mark: |
| ``rng`` | × |  :heavy_check_mark: |
| ``rsyslog`` | × |  :heavy_check_mark: |
| ``scaphandre`` | × |  :heavy_check_mark: |
| ``smartd`` || :heavy_check_mark: :exclamation:  <br>  smartd service has a condition set: ``ConditionVirtualization=no``and will only start if the system is not virtualized. Hence it can't be tested if the service is up and running.  Cf. ``smartd.py`` 41-53. |
| ``squid`` | × |  :heavy_check_mark: |
| ``tang`` || :question: <br> Service doesn't start. It appears that the units for the service have changed and are not implemented in the role setup yet. Is this role deprecated?|
| ``thanos_sidecar`` :new: || :heavy_check_mark: |
| ``traefik`` | × |  :heavy_check_mark: |
| ``tuned`` | × |  :heavy_check_mark: |
| ``virtualbmc`` | × |  :heavy_check_mark: |
| ``wireguard``| × |  :heavy_check_mark: |
| ``zuul`` || :question: <br> ``TASK [zuul : Copy pem files (e.g. for github)] "rsync: "/home/zuul/src/github.com/osism/ansible-collection-services/molecule/delegated/pem-files\" failed: No such file or directory`` Problem is ``src: "{{ playbook_dir }}/pem-files/"`` in ``copy_task.yml:4`` |

### CODE REVIEW

**Comment by gndrmnn** [comment]
recheck

File: molecule/delegated/tests/smartd.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    smartd_package = host.package(get_variable(host, "smartd_package_name"))
+    assert smartd_package.is_installed
+
+
+def test_smartd_logs(host):
+    smartd_service_name = get_variable(host, "smartd_service_name")
+    command = f'journalctl -u {smartd_service_name} --no-pager --since "1 minute ago"'
+    result = host.run(command)
+    assert result.rc == 0, "Failed to run journalctl for smartd service"
+    assert "error" not in result.stdout.lower(), "Error found in smartd service logs"
```
Comment (gndrmnn) [inline molecule/delegated/tests/smartd.py:16]: ?

File: molecule/delegated/vars/keycloak.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#keycloak_traefik: true
+#keycloak_galera_backend_enable: true
+#kolla_internal_vip_address: "127.0.0.1"
+#keycloak_database_name: "keycloak"
+#keycloak_database_username: "admin"
+#keycloak_database_password: "password"
+#database_address: "127.0.0.1"
+#database_user: "mysql"
+#database_password: "test123"
+#database_port: "3306"
```
Comment (gndrmnn) [inline molecule/delegated/vars/keycloak.yml:15]: rem

File: molecule/delegated/vars/manager.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+operator_user: zuul
+operator_group: zuul
+
+# testing
+#ara_server_mariadb_tag: '10.11'
```
Comment (gndrmnn) [inline molecule/delegated/vars/manager.yml:6]: rem

File: molecule/delegated/vars/netdata.yml [hunk 4]
```diff
@@ -1 +1,3 @@
 ---
+# testing
+#netdata_host_type: server
```
Comment (gndrmnn) [inline molecule/delegated/vars/netdata.yml:3]: rem

File: molecule/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
 molecule==6.0.3
 Jinja2==3.1.2
 pytest==7.4.4
-pytest-testinfra==10.0.0
+pytest-testinfra==10.0.0
```
Comment (gndrmnn) [inline molecule/requirements.txt:4]: Missing newline or added newline?

File: roles/manager/handlers/main.yml [hunk 6]
```diff
@@ -50,6 +52,7 @@
   changed_when: true
   notify:
     - Copy osismclient bash completion script
+  failed_when: false # changed for testing
```
Comment (gndrmnn) [inline roles/manager/handlers/main.yml:55]: is this supposed to be still in here?

Comment (sbstnnmnn) [reply roles/manager/handlers/main.yml:55]: of course not ;)

File: molecule/delegated/tests/smartd.py (hunk 1 above)
Comment (sbstnnmnn) [reply molecule/delegated/tests/smartd.py:16]: test deleted

File: molecule/delegated/tests/netdata.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+def test_repository_configuration(host):
+    if get_variable(host, "netdata_configure_repository"):
+        repo_key_url = get_variable(host, "netdata_debian_repository_key")
+        repo = get_variable(host, "netdata_debian_repository")
+        debian_version = host.system_info.distribution_version
+
+        if debian_version < "22.04":
+            assert host.run(f"apt-key list | grep {repo_key_url}").rc == 0
+        else:
```
[the remark this answers is not in the document]
Comment (sbstnnmnn) [reply molecule/delegated/tests/netdata.py:19]: done

File: molecule/delegated/tests/netdata.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            gpg_key_file = "/etc/apt/trusted.gpg.d/netdata.asc"
+            assert host.file(gpg_key_file).exists
```
[the remark this answers is not in the document]
Comment (sbstnnmnn) [reply molecule/delegated/tests/netdata.py:21]: Refactored test:
```
def test_repository_configuration(host):
    if get_variable(host, "netdata_configure_repository"):
        repo = get_variable(host, "netdata_debian_repository")
        key_file = host.file("/etc/apt/trusted.gpg.d/netdata.asc")
        key_content = get_from_url(get_variable(host, "netdata_debian_repository_key"))
        assert key_file.exists
        assert key_file.content_string == key_content
        assert host.run(f"apt-cache policy | grep {repo.split(' ')[1]}").rc == 0
```

**Review by sbstnnmnn — APPROVED** [review]

**Pull request merged**